File: Source/AbidzarFrame.WCF.Service/DocumentConfiguration.vb
Imports System.Configuration

Public Class DocumentConfiguration
    Inherits ConfigurationSection

    Protected Shared instance As DocumentConfiguration = Nothing

    Public Shared Function GetInstance() As DocumentConfiguration
        If (instance Is Nothing) Then
            instance = CType(ConfigurationManager.GetSection("DocumentConfiguration"), DocumentConfiguration)
        End If
        Return instance
    End Function

    <ConfigurationProperty("ConfigFilePath", defaultvalue:="")> _
    Public Property ConfigFilePath As String
        Get
            Return Me("ConfigFilePath")
        End Get
        Set(ByVal value As String)
            Me("ConfigFilePath") = value
        End Set
    End Property

End Class

File: Source/AbidzarFrame.API.Service/App_Start/DocumentConfiguration.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using System.Web;

namespace AbidzarFrame.API.Service
{
    public class DocumentConfiguration : ConfigurationSection
    {
        protected static ApiConfiguration instance = null;

        public static ApiConfiguration GetInstance()
        {
            if ((instance == null))
                instance = (ApiConfiguration)ConfigurationManager.GetSection("DocumentConfiguration");
            return instance;
        }

        [ConfigurationProperty("ConfigFilePath", DefaultValue = "")]
        public string ConfigFilePath
        {
            get { return "" + this["ConfigFilePath"]; }
            set { this["ConfigFilePath"] = value; }
        }

    }


}